### ISSUE TITLE
model restore

### ISSUE THREAD

**Issue by james20141606** [magenta/ddsp#58]
hey, I have a question about model save and restore. As you said here:
 > Saving weights in checkpoint format because saved_model requires handling variable batch size, which some synths and effects can't.

Do you mean the model also saves some synths and effects' variables?
I am struggling with this because I'd like to do some transfer learning with a new encoder but would like to use the pre-trained model's decoder' weights and I found that you use tf.train.Checkpoint.restore to restore the whole model. And you can use trainer.restore(model_dir) to restore the model during training. But it seems to make it hard to restore part of the model's weights using this coding style. 
Is there a way to restore only part of the pretrained model (like decoder) in restore part and pass it to the new model's decoder? Another solution I can think up is to restore the whole model and replace all the parts except decoder, which seems really weird and might now work.

**Comment by james20141606** [comment]
Could you give a hint which parts besides encoder and decoder are stored in the checkpoint? Can I modify the code and store these different parts separately using tf.train.Checkpoint like:
```
tf.train.Checkpoint( ptimizer= optimizer,
                                 encoder=encoder
                                 decoder=decoder)
```
and then only store the part I need?

**Comment by jesseengel** [comment]
Sorry for the delay in response. The only other variables besides the encoder and the decoder is in the processor_group if it has a trainable reverb (Reverb(trainable=True)).

**Comment by james20141606** [comment]
Thank you for your reply! I have partially solved the problem to load only decoder using `tf.train.load_variable` and set weights to certain layers. But I have some related questions:
- full model restore problem
I trained the DDSP autoencoder and tried to restore the full model using some codes like:
```
gin.add_config_file_search_path('ddsp-master/ddsp/training/gin/')
gin_file = 'ddsp-master/ddsp/training/gin/models/ae_abs.gin'
with gin.unlock_config():
model = ddsp.training.models.Autoencoder()
model.restore(MODEL_DIR)
```
but the strange thing is that most of the time it does not work, I could not load the model weights and I could see that the model weights in some layers are still initialization values. But I could load some of the models correctly although I trained them using same manner. Did that ever happen to you? I also got some error if I use `operative_config-0.gin` file, sometimes it's about cumsum operation which appears in `operative_config-0.gin` but did not match anything in the model.
- pretrainedembeddingloss problem
Another problem I got is after I loaded the pretrained decoder, the pretrainedembeddingloss will become 0 even at the first step training. I checked the codes and I do feed in original audio and reconstructed audio to compute pretrained embedding loss. And I could see that these two audios are different, but the loss is zero. Do you have any ideas on why this happens?
<img width="1074" alt="截屏2020-04-0218 06 38" src="https://user-images.githubusercontent.com/14315324/78304428-b67e2880-750c-11ea-9474-3daa700c955e.png">

**Comment by jesseengel** [comment]
Ah, there have been some changes to the code as cumsum has gotten more
efficient so we removed the custom hack we had. You can either go back to a
previous version such as https://github.com/magenta/ddsp/releases/tag/v0.0.6 or
you can just comment out those lines from your gin file.
